### DISCUSSION TITLE
Doxygen for R3BRoot

### DISCUSSION THREAD

**Discussion by YanzhaoW** [R3BRootGroup/R3BRoot#1098] [General]
## Describe the solution you'd like

Generate doxygen documentation and host it in Github pages. This could be very helpful if people want to understand how the source code is structured. With the doxygen, we could also put the documentation of each public member function directly in the source code.

## Additional context

To host the doxygen files in Github pages, we need to enable the permission from the Settings in this repository.

## About Doxygen

Doxygen is the de-facto documentation tools for large c++ projects. This has been used by ROOT and FairRoot:

- ROOT: https://root.cern.ch/doc/master/index.html
- FairRoot: https://fairrootgroup.github.io/FairRoot/html/index.html

If it's ok, I could create a PR to generate the doxygen documentation in a CI process. When dev branch is updated, it automatically generates a new version of doxygen documentation. I just need the permission to be enabled in the Settings.

@jose-luis-rs @hapol

**Comment by hapol** [comment]
Hi @YanzhaoW,
It would be a very good idea. I did such implementation for an old code but it required quite a lot of modifications in the comments structure and some keywords allocated next to each function declaration or definition. Does it requires many modifications in our case?

**Comment by YanzhaoW** [comment]
Hi @hapol 

I don't think Doxygen "requires" users to put comments to each function declaration or definition. Doxygen generates html files where you can host it somewhere like a webpage. Without doing anything, in the webpage you could still see all classes, functions or namespaces defined in the project. The comments, which are called doxygen comments, are not mandatory. If you add those comments, they will be displayed as additional descriptions for each class/function.

So modifications are not required in our case.

**Reply by jose-luis-rs** [reply]
Yes, it is a good idea. Thanks
